core-blocks/paragraph/index.js

```diff
@@ -166,12 +179,7 @@ class ParagraphBlock extends Component {
 		return (
 			<Fragment>
 				<BlockControls>
-					<AlignmentToolbar
-						value={ align }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { align: nextAlign } );
-						} }
-					/>
+					{ toolbar( this.props, this.state ) }
```

Comment:
I believe the second argument is useless here? Since the state is local to a block I believe we shouldn't use it in an API that can be common to multiple components. If block authors need state shared between `edit` and `toolbar`, it doesn't make sense to use `toolbar`, they'll just embed `BlockControls` in `edit`

Reply:
Perhaps... I was looking at https://github.com/WordPress/gutenberg/pull/3535 where me and @iseulde came up with very similar approaches, and for the inspector it was said that the block state was needed too, so I figured that perhaps we should keep the signature of "reusable controls" the same?